src/data/repositories/DHISRepository.js

```diff
@@ -0,0 +1,134 @@
+import { apiUrl } from "../../webapp/app.js";
+
+export class DHISRepository {
+	get($resource, dataSetsIds) {
+		return $resource(
+			apiUrl + "/dataSets.json",
+			{},
+			{
+				get: {
+					method: "GET",
+					params: {
+						fields: fields,
+						filter: `id:in:[${dataSetsIds}]`,
+						paging: false,
+					},
+				},
+			}
+		).get();
+	}
+
+	getDataSets(dataSets) {
```

Comment:
This method applies a transformation to data sets, it makes no sense to live in a repository (typically in the entity/use case)